pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/StUnionAggregationFunction.java

```diff
@@ -0,0 +1,137 @@
+
+
+public class StUnionAggregationFunction extends BaseSingleInputAggregationFunction<Geometry, ByteArray> {
+
+  /**
+   * Constructor for the class.
+   *
+   * @param expression Expression to aggregate on.
+   */
+  public StUnionAggregationFunction(ExpressionContext expression) {
+    super(expression);
+  }
+
+  @Override
+  public AggregationFunctionType getType() {
+    return AggregationFunctionType.ST_UNION;
+  }
+
+  @Override
+  public void accept(AggregationFunctionVisitorBase visitor) {
+    visitor.visit(this);
+  }
+
+  @Override
+  public AggregationResultHolder createAggregationResultHolder() {
+    return new ObjectAggregationResultHolder();
+  }
+
+  @Override
+  public GroupByResultHolder createGroupByResultHolder(int initialCapacity, int maxCapacity) {
+    return new ObjectGroupByResultHolder(initialCapacity, maxCapacity);
+  }
+
+  @Override
+  public void aggregate(int length, AggregationResultHolder aggregationResultHolder,
+      Map<ExpressionContext, BlockValSet> blockValSetMap) {
+    byte[][] bytesArray = blockValSetMap.get(_expression).getBytesValuesSV();
+    Geometry geometry = aggregationResultHolder.getResult();
+    for (int i = 0; i < length; i++) {
+      Geometry value = GeometrySerializer.deserialize(bytesArray[i]);
+      geometry = geometry == null ? value : geometry.union(value);
+    }
+    aggregationResultHolder.setValue(geometry);
+  }
+
+  @Override
+  public void aggregateGroupBySV(int length, int[] groupKeyArray, GroupByResultHolder groupByResultHolder,
+      Map<ExpressionContext, BlockValSet> blockValSetMap) {
+    byte[][] bytesArray = blockValSetMap.get(_expression).getBytesValuesSV();
+    for (int i = 0; i < length; i++) {
+      int groupKey = groupKeyArray[i];
+      Geometry value = GeometrySerializer.deserialize(bytesArray[i]);
+      groupByResultHolder.setValueForKey(groupKey, groupByResultHolder.getResult(groupKey) == null ? value
+          : ((Geometry) groupByResultHolder.getResult(groupKey)).union(value));
+    }
+  }
+
+  @Override
+  public void aggregateGroupByMV(int length, int[][] groupKeysArray, GroupByResultHolder groupByResultHolder,
+      Map<ExpressionContext, BlockValSet> blockValSetMap) {
+    byte[][] bytesArray = blockValSetMap.get(_expression).getBytesValuesSV();
+    for (int i = 0; i < length; i++) {
+      Geometry value = GeometrySerializer.deserialize(bytesArray[i]);
+      for (int groupKey : groupKeysArray[i]) {
+        groupByResultHolder.setValueForKey(groupKey, groupByResultHolder.getResult(groupKey) == null ? value
+            : ((Geometry) groupByResultHolder.getResult(groupKey)).union(value));
+      }
+    }
+  }
+
+  @Override
+  public Geometry extractAggregationResult(AggregationResultHolder aggregationResultHolder) {
+    return aggregationResultHolder.getResult();
+  }
+
+  @Override
+  public Geometry extractGroupByResult(GroupByResultHolder groupByResultHolder, int groupKey) {
+    return groupByResultHolder.getResult(groupKey);
```

Comment:
Need to handle `null` here (return an empty Geometry)

Reply:
same question. There is no empty geometry, but emty geometry per specific type, e.g. `GEOMETRYCOLLECTION EMPTY`, `POINT EMPTY`, `LINESTRING EMPTY`

Reply:
We need to choose a default geometry to return when no record is selected. I think we can use empty `Point` as the default

Reply:
can you show me a query that will exhibit this no-record case?

Reply:
SELECT ... WHERE col = "no_match_ value"